File: src/test/java/howtoprogram/austen/test/HistogramTest.java
package howtoprogram.austen.test;

import howtoprogram.austen.Histogram;
import org.junit.Assert;
import org.junit.Test;

import java.util.Set;

public class HistogramTest {

    @Test
    public void emptyToStartWith() {
        Histogram histogram = new Histogram();
        Assert.assertEquals(0, histogram.allWords().size());
    }

    @Test
    public void countIsZeroForWordNotGiven() {
        Histogram histogram = new Histogram();
        Assert.assertEquals(0, histogram.numberOfTimesGiven("xylophone"));
    }

    @Test
    public void recordOneWord() {
        Histogram histogram = new Histogram();
        histogram.record("piano");
        Set<String> words = histogram.allWords();
        Assert.assertEquals(1, words.size());
        Assert.assertTrue(words.contains("piano"));
        Assert.assertEquals(1, histogram.numberOfTimesGiven("piano"));
    }

    @Test
    public void recordOneWordTwice() {
        Histogram histogram = new Histogram();
        histogram.record("piano");
        histogram.record("piano");
        Set<String> words = histogram.allWords();
        Assert.assertEquals(1, words.size());
        Assert.assertTrue(words.contains("piano"));
        Assert.assertEquals(1, histogram.numberOfTimesGiven("piano"));
    }

    @Test
    public void recordTwoWords() {
        Histogram histogram = new Histogram();
        histogram.record("piano");
        histogram.record("violin");
        Set<String> words = histogram.allWords();
        Assert.assertEquals(2, words.size());
        Assert.assertTrue(words.contains("piano"));
        Assert.assertEquals(1, histogram.numberOfTimesGiven("piano"));
        Assert.assertTrue(words.contains("violin"));
        Assert.assertEquals(1, histogram.numberOfTimesGiven("violin"));
    }

}

File: src/test/kotlin/pfb/austen/test/HistogramTest.kt
package pfb.austen.test

import org.junit.Assert
import org.junit.Test
import pfb.austen.Histogram

public class HistogramTest {
    @Test
    fun emptyToStartWith() {
        val histogram = Histogram()
        Assert.assertEquals(0, histogram.allWords().size)
    }

    @Test
    fun unknownWord() {
        val histogram = Histogram()
        val given = histogram.numberOfTimesGiven("xylophone")
        Assert.assertEquals(0, given)
    }

    @Test
    fun recordOneWord() {
        val histogram = Histogram()
        histogram.record("piano")
        val words = histogram.allWords()
        Assert.assertEquals(1, words.size)
        Assert.assertTrue(words.contains("piano"))
        Assert.assertEquals(1, histogram.numberOfTimesGiven("piano"))
    }

    @Test
    fun recordOneWordTwice() {
        val histogram = Histogram()
        histogram.record("piano")
        histogram.record("piano")
        val words = histogram.allWords()
        Assert.assertEquals(1, words.size)
        Assert.assertTrue(words.contains("piano"))
        Assert.assertEquals(2, histogram.numberOfTimesGiven("piano"))
    }

    @Test
    fun recordTwoWords() {
        val histogram = Histogram()
        histogram.record("piano")
        histogram.record("violin")
        val words = histogram.allWords()
        Assert.assertEquals(2, words.size.toLong())
        Assert.assertTrue(words.contains("piano"))
        Assert.assertTrue(words.contains("violin"))
        Assert.assertEquals(1, histogram.numberOfTimesGiven("piano"))
        Assert.assertEquals(1, histogram.numberOfTimesGiven("violin"))
    }
}